### PR TITLE
Update client to handle empty JSON responses

### PR DESCRIPTION
**Pull request by james-harper**
### Changed
- Updated ApiClient implementations so that they can handle empty JSON responses

### CODE REVIEW

File: src/ApiClient.php [hunk 1]
```diff
@@ -38,7 +38,13 @@ protected function processRequestBody(array $body)
...
     protected function processResponse(ResponseInterface $response)
     {
-        if ($responseBody = json_decode($response->getBody()->getContents(), true)) {
+        if ($response->getStatusCode() == 204) {
+            return [];
+        }
+
+        $responseBody = json_decode($response->getBody()->getContents(), true);
+
+        if (!is_null($responseBody)) {
```
Comment (wnowicki) [inline src/ApiClient.php]:
```php
if (is_array($responseBody)) {
```

as that's what you are looking for to return

File: src/RetryApiClient.php [hunk 2]
```diff
@@ -74,7 +74,13 @@ protected function processRequestBody(array $body)
...
     protected function processResponse(ResponseInterface $response)
     {
-        if ($responseBody = json_decode($response->getBody()->getContents(), true)) {
+        if ($response->getStatusCode() == 204) {
+            return [];
+        }
+
+        $responseBody = json_decode($response->getBody()->getContents(), true);
+
+        if (!is_null($responseBody)) {
```
Comment (wnowicki) [inline src/RetryApiClient.php]: the same ;)

**Comment by james-harper** [comment]
@wnowicki An issue I've run into while using this ApiClient is that it did not handle empty json responses (204s)  well. It would throw a "body is malformed JSON" exception. Updated to handle these better

File: src/RetryApiClient.php (hunk 2 above)
Comment (james-harper) [inline src/RetryApiClient.php]: Now checking if response decodes to an array

**Pull request merged**